src/5e-SRD-Spells.json

```diff
@@ -707,13 +710,15 @@
       "You touch a closed door, window, gate, chest, or other entryway, and it becomes locked for the duration. You and the creatures you designate when you cast this spell can open the object normally. You can also set a password that, when spoken within 5 feet of the object, suppresses this spell for 1 minute. Otherwise, it is impassable until it is broken or the spell is dispelled or suppressed. Casting knock on the object suppresses arcane lock for 10 minutes.",
       "While affected by this spell, the object is more difficult to break or force open; the DC to break it or pick any locks on it increases by 10."
     ],
-    "range": "Touch",
+    "range": {"value": 0, "unit": "ft"},
+    "range_type": "touch",
     "components": ["V", "S", "M"],
     "material": "Gold dust worth at least 25gp, which the spell consumes.",
     "ritual": false,
-    "duration": "Until dispelled",
+    "duration": {"value": -1, "unit": "sec"},
```

Comment:
I'm not a huge fan of `-1` representing "infinity" or "until dispelled". It doesn't scan well and would be less intuitive, imho, to handle computationally. Maybe it would be more appropriate to indicate this kind of behaviour in the unit? Or, given the use of `duration_type`, couldn't `duration` simply be omitted for certain `duration_type`s, rather than supplying data that looks valid but patterns differently to the rest?

```json
{
  "value": 1,
  "unit": "infinity",
  "up_to": true
}
```

On a related note, is `sec` ever used anywhere besides "Instantaneous" (0 sec) and "Until dispelled" (-1 sec)? Perhaps "Instantaneous" out to be replaced with it's own unit, or a `duration_type` that lacks a `duration`, also?

Reply:
Yeah, using `duration_type` where a unit is not specifiable should be better imo

Reply:
I think the `duration_type` without a `duration` makes sense for the `instantaneous` and `until something happens` cases. We might want to define a model or enum in the API to make it clear to a consumer what values can be expected in those cases.